star-wars-films/src/PersonDetail/PersonDetail.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react';
+import './PersonDetail.css';
+import PropTypes from 'prop-types';
+
+class PersonDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      person: {}
+    }
+  }
+
+  componentDidMount() {
+    const peopleId = this.props.match.params.peopleId;
+    fetch(`https://swapi.co/api/people/${peopleId}/`)
+      .then(response => response.json())
+      .then(data => {
+        this.setState((prevState, props) => {
+          return {
+            person: data
+          };
+        });
+      })
+      .catch(error => console.log(error));
+  };
+
+  getFilmList(films) {
+    const filmNames = ["A New Hope", "The Empire Strikes Back", "Return of the Jedi", "The Phantom Menace", "Attack of the Clones", "Revenge of the Sith", "The Force Awakens", "The Last Jedi"];
+    let finalFilmList = '';
+    for (let i = 0; i < films.length; i++){
+      const url = films[i];
+      let filmNum = url.split('/').reverse()[1];
+      filmNum = filmNum - 1;
+      finalFilmList += filmNames[filmNum];
+      if (i < (films.length - 1)) {
+        finalFilmList += ", ";
+      }
+    }
+    return (
+      <span>
+        {finalFilmList}
+      </span>
+    );
+  };
+  // // Can't do below as return statement in getFilmList
+  // // tells me that finalFilmList.map is not a function;
+  // // tried finalFilmList.forEach and got the same error.
+  //
+  // {finalFilmList.map((finalFilmList, idx) => {
+  //    return <span>{finalFilmList}, </span>
+  // })}
+
+  getHomePlanet(planet) {
+    const planetNames = ["Tatooine","Alderaan","Yavin IV","Hoth","Dagobah","Bespin","Endor","Naboo","Coruscant", "Kamino", "Geonosis", "Utapau", "Mustafar", "Kashyyyk", "Polis Massa", "Mygeeto", "Felucia", "Cato Neimoidia", "Saleucami", "Stewjon", "Eriadu", "Corellia", "Rodia", "Nal Hutta", "Dantooine", "Bestine IV", "Ord Mantell", "unknown", "Trandosha", "Socorro", "Mon Cala", "Chandrila", "Sullust", "Toydaria", "Malastare", "Dathomir", "Ryloth", "Aleen Minor", "Vulpter", "Troiken", "Tund", "Haruun Kal", "Cerea", "Glee Anselm", "Iridonia", "Tholoth", "Iktotch", "Quermia", "Dorin", "Champala", "Mirial", "Serenno", "Concord Dawn","Zolan","Ojom","Skako","Muunilinst","Shili","Kalee","Umbara","Jakku"]
```

Comment:
Like with the film data above. I kind of faked this without doing a second API call. I'd be curious how to do second call inside this file.

Reply:
I'd add planets[] to state in home.js and get that list in home.js componentDidMount.